client/src/components/Profile/Profile.js

```diff
@@ -0,0 +1,306 @@
+import React from "react";
+import "./Profile.css";
+import star from "./Images/star.png";
+import { format, startOfWeek, endOfWeek, addDays } from "date-fns";
+import ScheduleSelector from "react-schedule-selector";
+let currentClicked = null;
+let currentClickedID = null;
+
+function Timeslot(props) {
+  return (
+    <div
+      className="item-cell"
+      style={{ backgroundColor: props.color }}
+      id={props.id}
+      onClick={props.onClick}
+    ></div>
+  );
+}
+
+class Shift {
+  constructor(color, id, start, end, day) {
+    this.color = color;
+    this.id = id;
+    this.start = start;
+    this.end = end;
+    this.day = day;
+  }
+}
+
+function initialShifts() {
+  let a = [];
+  for (var i = 0; i < 168; i += 1) {
+    a.push(new Shift("#f8f8f8", null, null, null, null));
+  }
+  let count = 0;
+  for (var i = 0; i <= 23; i += 1) {
+    for (var j = 0; j <= 6; j += 1) {
+      a[count].start = i;
+      a[count].end = i + 1;
+      a[count].day = j;
+      count += 1;
+    }
+  }
+  return a;
+}
+
+var currentDate = new Date();
+var weekString =
+  format(currentDate, "MMMM") +
+  " " +
+  format(currentDate, "YYYY") +
+  ": " +
+  format(startOfWeek(currentDate), "MM/DD") +
+  " - " +
+  format(endOfWeek(currentDate), "MM/DD");
+
+export default class Profile extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { shifts: initialShifts(), schedule: [] };
+    this.currentDate = new Date();
+    this.deselectCell = <div class="deselectCell"></div>;
+    this.selectCell = <div class="selectCell"></div>;
+    this.stateFixer = this.stateFixer.bind(this);
+    this.submitClick = this.submitClick.bind(this);
+    this.cancelClick = this.cancelClick.bind(this);
+    this.renderCustomDateCell = this.renderCustomDateCell.bind(this);
+  }
+
+  componentDidMount() {
+    fetch("/availability/" + this.props.match.params.userId)
+      .then((response) => {
+        console.log("availability response");
+        return response.json();
+      })
+      .then((jsonResponse) => {
+        console.log("availability response 2");
+        this.setState({ schedule: jsonResponse.schedule });
+      });
+  }
+
+  stateFixer(e) {
+    if (
+      e.target.id != "" &&
+      e.target.style.backgroundColor != "rgb(193, 135, 211)"
+    ) {
+      /*The point of the background color check is to make sure that once a shift is requested to be covered,
+       * this can't happen again for the same shift.
+       */
+      currentClicked = e;
+      currentClickedID = e.target.id;
+      this.openModal();
+    }
+  }
+
+  renderCustomDateCell = (time, selected, innerRef) => {
+    return (
+      <div style={{ textAlign: "center" }} ref={innerRef}>
+        {selected ? this.selectCell : this.deselectCell}
+      </div>
+    );
+  };
+
+  submitClick() {
```

Comment:
We can delete this function since there's nothing to submit anymore. Delete it in the button where it's being called too (and delete the button).